File: src/main/java/com/mmust/token/DivisionOpToken.java
package com.mmust.token;

public class DivisionOpToken extends OperatorToken {

	public DivisionOpToken() {
		super("/");
	}

	@Override
	public double eval(double op1, double op2) {
		return op1 / op2;
	}

	@Override
	public int getPrecedence() {
		return 3;
	}

	@Override
	public Associativity getAssociativity() {
		return Associativity.LEFT;
	}
}

File: src/main/kotlin/com/mmust/token/DivisionOpToken.kt
package com.mmust.token

class DivisionOpToken : OperatorToken("/") {

    override val precedence: Int
        get() = 3

    override val associativity: Associativity
        get() = Associativity.LEFT

    override fun eval(op1: Double, op2: Double): Double {
        return op1 / op2
    }
}
